<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Deconvolute bulk RNA-Seq samples using a pre-trained DigitalDLSorter model — deconvDigitalDLSorter • digitalDLSorteR</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Deconvolute bulk RNA-Seq samples using a pre-trained DigitalDLSorter model — deconvDigitalDLSorter" />
<meta property="og:description" content="Deconvolute bulk gene expression samples (bulk RNA-Seq) to enumerate and
quantify the proportion of cell types present in a bulk sample using Deep
Neural Network models. This function is intended for users who want to use
pre-trained models integrated in the package. So far, the available models
allow to deconvolute the immune infiltration of breast cancer (using data from
Chung et al., 2017) and the immune infiltration of colorectal cancer (using
data from Li et al., 2017) samples. For the former, two models are available
at two different levels of specificity: specific cell types
(breast.chung.specific) and generic cell types
(breast.chung.generic). See breast.chung.generic,
breast.chung.specific, and colorectal.li documentation from the
digitalDLSorteRdata package for more details." />
<meta property="og:image" content="/logo.png" />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">digitalDLSorteR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href=".././articles/pretrainedModels.html">Using pre-trained context-specific deconvolution models</a>
    </li>
    <li>
      <a href=".././articles/newModels.html">Building new deconvolution models</a>
    </li>
    <li>
      <a href=".././articles/realModelExample.html">Performance of a real model: deconvolution of colorectal cancer samples</a>
    </li>
    <li>
      <a href=".././articles/kerasIssues.html">Keras/TensorFlow installation and configuration</a>
    </li>
    <li>
      <a href=".././articles/hdf5Backend.html">HDF5 files as back-end</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/diegommcc/digitalDLSorteR">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Deconvolute bulk RNA-Seq samples using a pre-trained DigitalDLSorter model</h1>
    <small class="dont-index">Source: <a href='https://github.com/diegommcc/digitalDLSorteR/R/dnnModel.R'><code>R/dnnModel.R</code></a></small>
    <div class="hidden name"><code>deconvDigitalDLSorter.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Deconvolute bulk gene expression samples (bulk RNA-Seq) to enumerate and
quantify the proportion of cell types present in a bulk sample using Deep
Neural Network models. This function is intended for users who want to use
pre-trained models integrated in the package. So far, the available models
allow to deconvolute the immune infiltration of breast cancer (using data from
Chung et al., 2017) and the immune infiltration of colorectal cancer (using
data from Li et al., 2017) samples. For the former, two models are available
at two different levels of specificity: specific cell types
(<code>breast.chung.specific</code>) and generic cell types
(<code>breast.chung.generic</code>). See <code>breast.chung.generic</code>,
<code>breast.chung.specific</code>, and <code>colorectal.li</code> documentation from the
<span class="pkg">digitalDLSorteRdata</span> package for more details.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>deconvDigitalDLSorter</span><span class='op'>(</span>
  <span class='va'>data</span>,
  model <span class='op'>=</span> <span class='cn'>NULL</span>,
  batch.size <span class='op'>=</span> <span class='fl'>128</span>,
  normalize <span class='op'>=</span> <span class='cn'>TRUE</span>,
  simplify.set <span class='op'>=</span> <span class='cn'>NULL</span>,
  simplify.majority <span class='op'>=</span> <span class='cn'>NULL</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p>Matrix or data frame with bulk RNA-Seq samples with genes as rows
in SYMBOL notation and samples as columns.</p></td>
    </tr>
    <tr>
      <th>model</th>
      <td><p>Pre-trained DNN model to use to deconvolute <code>data</code>. Up to
now, the available models are intended to deconvolute samples from breast
cancer (<code>breast.chung.generic</code> and <code>breast.chung.specific</code>) and
colorectal cancer (<code>colorectal.li</code>). These pre-trained models are
stored in the <span class="pkg">digitalDLSorteRdata</span> package, so it must be installed
together with <span class="pkg">digitalDLSorteR</span> to use this function.</p></td>
    </tr>
    <tr>
      <th>batch.size</th>
      <td><p>Number of samples loaded into RAM each time during the
deconvolution process. If not specified, <code>batch.size</code> will be set to
128.</p></td>
    </tr>
    <tr>
      <th>normalize</th>
      <td><p>Normalize data before deconvolution (<code>TRUE</code> by default).</p></td>
    </tr>
    <tr>
      <th>simplify.set</th>
      <td><p>List specifying which cell types should be compressed into
a new label whose name will be the list name item. See examples and
vignettes for details.</p></td>
    </tr>
    <tr>
      <th>simplify.majority</th>
      <td><p>List specifying which cell types should be compressed
into the cell type with the highest proportion in each sample. Unlike
<code>simplify.set</code>, this argument allows to maintain the complexity of the
results while compressing the information, as no new labels are created.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Show informative messages during execution.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A data frame with samples (\(i\)) as rows and cell types (\(j\))
as columns. Each entry represents the predicted proportion of cell type
\(j\) in sample \(i\).</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function is intended for users who want to use <span class="pkg">digitalDLSorteR</span> to
deconvolute their bulk RNA-Seq samples using pre-trained models. For users who
want to build their own models from other scRNA-Seq datasets, see the
<code><a href='loadSCProfiles.html'>loadSCProfiles</a></code> and <code><a href='deconvDigitalDLSorterObj.html'>deconvDigitalDLSorterObj</a></code>
functions.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Chung, W., Eum, H. H., Lee, H. O., Lee, K. M., Lee, H. B., Kim, K.
T., et al. (2017). Single-cell RNA-seq enables comprehensive tumour and
immune cell profiling in primary breast cancer. Nat. Commun. 8 (1), 15081.
doi: doi: <a href='https://doi.org/10.1038/ncomms15081'>10.1038/ncomms15081</a>
.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='deconvDigitalDLSorterObj.html'>deconvDigitalDLSorterObj</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'><span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/ns-load.html'>requireNamespace</a></span><span class='op'>(</span><span class='st'>"digitalDLSorteRdata"</span>, quietly <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'>  <span class='co'># to ensure compatibility</span></span>
<span class='r-in'>  <span class='fu'>tensorflow</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/tensorflow/man/tf.html'>tf</a></span><span class='op'>$</span><span class='va'>compat</span><span class='op'>$</span><span class='va'>v1</span><span class='op'>$</span><span class='fu'>disable_eager_execution</span><span class='op'>(</span><span class='op'>)</span></span>
<span class='r-in'>  <span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/diegommcc/digitalDLSorteRdata'>digitalDLSorteRdata</a></span><span class='op'>)</span></span>
<span class='r-in'>  <span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>breast.chung.specific</span><span class='op'>)</span></span>
<span class='r-in'>  <span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>TCGA.breast.small</span><span class='op'>)</span></span>
<span class='r-in'>  <span class='va'>results1</span> <span class='op'>&lt;-</span> <span class='fu'>deconvDigitalDLSorter</span><span class='op'>(</span></span>
<span class='r-in'>    data <span class='op'>=</span> <span class='va'>TCGA.breast.small</span>,</span>
<span class='r-in'>    model <span class='op'>=</span> <span class='va'>breast.chung.specific</span>,</span>
<span class='r-in'>    normalize <span class='op'>=</span> <span class='cn'>TRUE</span></span>
<span class='r-in'>  <span class='op'>)</span></span>
<span class='r-in'>  <span class='co'># simplify arguments</span></span>
<span class='r-in'>  <span class='va'>simplify</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Tumor <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ER+"</span>, <span class='st'>"HER2+"</span>, <span class='st'>"ER+/HER2+"</span>, <span class='st'>"TNBC"</span><span class='op'>)</span>,</span>
<span class='r-in'>                   Bcells <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Bmem"</span>, <span class='st'>"BGC"</span><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'>  <span class='co'># in this case the names of the list will be the new labels</span></span>
<span class='r-in'>  <span class='va'>results2</span> <span class='op'>&lt;-</span> <span class='fu'>deconvDigitalDLSorter</span><span class='op'>(</span></span>
<span class='r-in'>    <span class='va'>TCGA.breast.small</span>,</span>
<span class='r-in'>    model <span class='op'>=</span> <span class='va'>breast.chung.specific</span>,</span>
<span class='r-in'>    normalize <span class='op'>=</span> <span class='cn'>TRUE</span>,</span>
<span class='r-in'>    simplify.set <span class='op'>=</span> <span class='va'>simplify</span></span>
<span class='r-in'>  <span class='op'>)</span></span>
<span class='r-in'>  <span class='co'># in this case the cell type with the highest proportion will be the new label</span></span>
<span class='r-in'>  <span class='va'>results3</span> <span class='op'>&lt;-</span> <span class='fu'>deconvDigitalDLSorter</span><span class='op'>(</span></span>
<span class='r-in'>    <span class='va'>TCGA.breast.small</span>,</span>
<span class='r-in'>    model <span class='op'>=</span> <span class='va'>breast.chung.specific</span>,</span>
<span class='r-in'>    normalize <span class='op'>=</span> <span class='cn'>TRUE</span>,</span>
<span class='r-in'>    simplify.majority <span class='op'>=</span> <span class='va'>simplify</span></span>
<span class='r-in'>  <span class='op'>)</span></span>
<span class='r-in'><span class='op'>}</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Filtering 32876 features in data that are not present in trained model</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Setting 11305 features that are not present in trained model to zero</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Normalizing data</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Predicting cell types present in the provided samples</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>`predict_generator` is deprecated. Use `predict` instead, it now accept generators.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> DONE</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Filtering 32876 features in data that are not present in trained model</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Setting 11305 features that are not present in trained model to zero</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Normalizing data</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Predicting cell types present in the provided samples</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>`predict_generator` is deprecated. Use `predict` instead, it now accept generators.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> DONE</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Filtering 32876 features in data that are not present in trained model</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Setting 11305 features that are not present in trained model to zero</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Normalizing data</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === Predicting cell types present in the provided samples</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>`predict_generator` is deprecated. Use `predict` instead, it now accept generators.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> DONE</span>
<span class='r-in'></span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by <a href="https://github.com/diegommcc" class="external-link">Diego Mañanes</a>, <a href="https://www.cnic.es/en/carlos-torroja" class="external-link">Carlos Torroja</a>, <a href="https://www.cnic.es/en/fatima-sanchez-cabo" class="external-link">Fatima Sanchez-Cabo</a>.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


