<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Show distribution plots of the cell proportions generated by
generateBulkCellMatrix — showProbPlot • digitalDLSorteR</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Show distribution plots of the cell proportions generated by
generateBulkCellMatrix — showProbPlot" />
<meta property="og:description" content="Show distribution plots of the cell proportions generated by
generateBulkCellMatrix. These frequencies will determine the
proportion of different cell types used during the simulation of pseudo-bulk
RNA-Seq samples. There are 6 subsets of proportions generated by different
approaches that can be visualized in three ways: box plots, violin plots and
lines plots. You can also plot the probabilities based on the number of
different cell types present in the samples by setting type.plot =
'nCellTypes'." />
<meta property="og:image" content="/logo.png" />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">digitalDLSorteR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href=".././articles/pretrainedModels.html">Using pre-trained context-specific deconvolution models</a>
    </li>
    <li>
      <a href=".././articles/newModels.html">Building new deconvolution models</a>
    </li>
    <li>
      <a href=".././articles/realModelExample.html">Performance of a real model: deconvolution of colorectal cancer samples</a>
    </li>
    <li>
      <a href=".././articles/kerasIssues.html">Keras/TensorFlow installation and configuration</a>
    </li>
    <li>
      <a href=".././articles/hdf5Backend.html">HDF5 files as back-end</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/diegommcc/digitalDLSorteR">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Show distribution plots of the cell proportions generated by
<code><a href='generateBulkCellMatrix.html'>generateBulkCellMatrix</a></code></h1>
    <small class="dont-index">Source: <a href='https://github.com/diegommcc/digitalDLSorteR/R/utils.R'><code>R/utils.R</code></a></small>
    <div class="hidden name"><code>showProbPlot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Show distribution plots of the cell proportions generated by
<code><a href='generateBulkCellMatrix.html'>generateBulkCellMatrix</a></code>. These frequencies will determine the
proportion of different cell types used during the simulation of pseudo-bulk
RNA-Seq samples. There are 6 subsets of proportions generated by different
approaches that can be visualized in three ways: box plots, violin plots and
lines plots. You can also plot the probabilities based on the number of
different cell types present in the samples by setting <code>type.plot =
'nCellTypes'</code>.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>showProbPlot</span><span class='op'>(</span><span class='va'>object</span>, <span class='va'>type.data</span>, <span class='va'>set</span>, type.plot <span class='op'>=</span> <span class='st'>"boxplot"</span><span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p><code><a href='DigitalDLSorter-class.html'>DigitalDLSorter</a></code> object with
<code>prob.cell.types</code> slot with <code>plot</code> slot.</p></td>
    </tr>
    <tr>
      <th>type.data</th>
      <td><p>Subset of data to show: <code>train</code> or <code>test</code>.</p></td>
    </tr>
    <tr>
      <th>set</th>
      <td><p>Integer determining which of the 6 different subsets to display.</p></td>
    </tr>
    <tr>
      <th>type.plot</th>
      <td><p>Character determining which type of visualization to
display. It can be <code>'boxplot'</code>, <code>'violinplot'</code>, <code>'linesplot'</code> or
<code>'ncelltypes'</code>. See Description for more information.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>ggplot</code> object.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>These plots are only for diagnostic purposes. This is the reason because they
are generated without any parameter introduced by the user.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='generateBulkCellMatrix.html'>generateBulkCellMatrix</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'><span class='co'># simulating data</span></span>
<span class='r-in'><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'>SingleCellExperiment</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment</a></span><span class='op'>(</span></span>
<span class='r-in'>  <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span></span>
<span class='r-in'>    <span class='fu'><a href='https://rdrr.io/r/stats/Poisson.html'>rpois</a></span><span class='op'>(</span><span class='fl'>100</span>, lambda <span class='op'>=</span> <span class='fl'>5</span><span class='op'>)</span>, nrow <span class='op'>=</span> <span class='fl'>40</span>, ncol <span class='op'>=</span> <span class='fl'>30</span>, </span>
<span class='r-in'>    dimnames <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"Gene"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>40</span><span class='op'>)</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"RHC"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>30</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'>  <span class='op'>)</span>,</span>
<span class='r-in'>  colData <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span class='r-in'>    Cell_ID <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"RHC"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>30</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span class='r-in'>    Cell_Type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"CellType"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>4</span><span class='op'>)</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>30</span>, </span>
<span class='r-in'>                       replace <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span class='r-in'>  <span class='op'>)</span>,</span>
<span class='r-in'>  rowData <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span class='r-in'>    Gene_ID <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"Gene"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>40</span><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'>  <span class='op'>)</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'><span class='va'>DDLS</span> <span class='op'>&lt;-</span> <span class='fu'><a href='loadSCProfiles.html'>loadSCProfiles</a></span><span class='op'>(</span></span>
<span class='r-in'>  single.cell.data <span class='op'>=</span> <span class='va'>sce</span>,</span>
<span class='r-in'>  cell.ID.column <span class='op'>=</span> <span class='st'>"Cell_ID"</span>,</span>
<span class='r-in'>  gene.ID.column <span class='op'>=</span> <span class='st'>"Gene_ID"</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'><span class='va'>probMatrix</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span class='r-in'>  Cell_Type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"CellType"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>4</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span class='r-in'>  from <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span>, <span class='fl'>1</span>, <span class='fl'>30</span><span class='op'>)</span>,</span>
<span class='r-in'>  to <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>15</span>, <span class='fl'>15</span>, <span class='fl'>50</span>, <span class='fl'>70</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'><span class='va'>DDLS</span> <span class='op'>&lt;-</span> <span class='fu'><a href='generateBulkCellMatrix.html'>generateBulkCellMatrix</a></span><span class='op'>(</span></span>
<span class='r-in'>  object <span class='op'>=</span> <span class='va'>DDLS</span>,</span>
<span class='r-in'>  cell.ID.column <span class='op'>=</span> <span class='st'>"Cell_ID"</span>,</span>
<span class='r-in'>  cell.type.column <span class='op'>=</span> <span class='st'>"Cell_Type"</span>,</span>
<span class='r-in'>  prob.design <span class='op'>=</span> <span class='va'>probMatrix</span>,</span>
<span class='r-in'>  num.bulk.samples <span class='op'>=</span> <span class='fl'>60</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> === The number of bulk RNA-Seq samples that will be generated is equal to 60</span>
<span class='r-err co'><span class='r-pr'>#&gt;</span> <span class='error'>Error in generateBulkCellMatrix(object = DDLS, cell.ID.column = "Cell_ID",     cell.type.column = "Cell_Type", prob.design = probMatrix,     num.bulk.samples = 60)</span> Not all cell types considered in DigitalDLSorter object are in test data. digitalDLSorteR needs to have all cell types in both subsets (training and test). Please, provide a bigger single-cell experiment or consider simulate new single-cell profiles with 'estimateZinbwaveParams' and 'simSCProfiles' functions</span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span></span>
<span class='r-in'>  X <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>6</span>, FUN <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'>      <span class='fu'>showProbPlot</span><span class='op'>(</span></span>
<span class='r-in'>        <span class='va'>DDLS</span>,</span>
<span class='r-in'>        type.data <span class='op'>=</span> <span class='st'>"train"</span>,</span>
<span class='r-in'>        set <span class='op'>=</span> <span class='va'>x</span>,</span>
<span class='r-in'>        type.plot <span class='op'>=</span> <span class='st'>"boxplot"</span></span>
<span class='r-in'>      <span class='op'>)</span></span>
<span class='r-in'>    <span class='op'>}</span></span>
<span class='r-in'>  <span class='op'>)</span></span>
<span class='r-err co'><span class='r-pr'>#&gt;</span> <span class='error'>Error in showProbPlot(DDLS, type.data = "train", set = x, type.plot = "boxplot")</span> 'prob.cell.types' slot is empty</span>
<span class='r-in'>  </span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by <a href="https://github.com/diegommcc" class="external-link">Diego Mañanes</a>, <a href="https://www.cnic.es/en/carlos-torroja" class="external-link">Carlos Torroja</a>, <a href="https://www.cnic.es/en/fatima-sanchez-cabo" class="external-link">Fatima Sanchez-Cabo</a>.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


